src/ui/grid.cpp

```diff
@@ -129,6 +129,26 @@ void Grid::setStyle(Style* style)
   setGap(style->gap());
 }
 
+void Grid::setColumns(int columns)
+{
+  if (columns == m_colstrip.size()) return;
+
+  int oldSize = m_colstrip.size();
+
+  m_colstrip.resize(columns);
+  for (std::size_t col=0; col<m_colstrip.size(); ++col) {
+    m_colstrip[col].size = 0;
+    m_colstrip[col].expand_count = 0;
+  }
```

Comment:
No sure if we should reset the information of the columns after the resize or doing something similar as the rows (I didn't test this).

Reply:
I have just tried removing the for loop, and it seems to work anyway. So, I've removed it. Thanks!